Add Area, Device, and Floor options to tile-card state content

pcan08 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue or discussion
  in the additional information section.
-->

Add Area, Device, and Floor options in state content options list for tile card. 

https://github.com/user-attachments/assets/02fbb7d7-099c-46e0-b839-e09ef1efea46

This idea comes from a discussion (see link below) which I found very interesting. 
It complements or perhaps even replaces PR #26722. 
~~You may consider to have the same for device name and floor name. If you accept this PR, I candidates to continue working on these additional options, let me know.~~

The new options are put in the list of state content options only if relevant depending the entity/device config.

**Note** 
`ha-entity-state-content-picker` have a new properties `allowArea`, `allowFloor` and `allowDevice` to allow or not adding these options so that it can be used in other card editors (`hui-heading-entity-editor` and `hui-entity-badge-editor`) if desire.
That follows the current design used to have entity name in the picker option with `allowName` for the components mentioned above.

**What else:**
I also started a new branch not yet publish where I added an indicator to the chip in the editor if the `state_content` config param contains a value that the entity does support (no area / floor set for example). If you're interested, i can start a new PR or merge the branch on this one. That looks like this:
<img width="504" height="450" alt="Capture d’écran du 2025-09-04 19-10-53" src="https://github.com/user-attachments/assets/dc81bdcb-038d-4c3e-9ab0-d6143051bbe3" />


## Type of change
<!--
  What type of change does your PR introduce to the Home Assistant frontend?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR.
-->

```yaml
type: tile
entity: fan.ventilateur_bureau
features_position: bottom
vertical: false
state_content:
  - direction
  - area
```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: https://github.com/orgs/home-assistant/discussions/869
- Link to documentation pull request:

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

<!--
  Thank you for contributing <3
-->

[docs-repository]: https://github.com/home-assistant/home-assistant.io

pcan08 committed:
Add area in state content selector

pcan08 committed:
Hide area option if entity does not have area

pcan08 committed:
Add allowArea property to ha-entity-state-content-picker

This way component using state content selector can be configure to have area option.
For now only allowed for tile-card.

pcan08 committed:
fix missing translation

pcan08 committed:
Merge branch 'dev' into area-in-state-content

pcan08 committed:
Merge branch 'dev' into area-in-state-content

piitaya commented:
HI 👋 I was currently working on a similar PR than https://github.com/home-assistant/frontend/pull/26722. I'll push a PR soon and will ask you for a review. 
I was waiting for this PR be merged before opening the name preset PR. https://github.com/home-assistant/frontend/pull/26057

Then I think we can move forward with this one. It is also a good option to let user adding area, floor, device name in state content.  I'll keep you updated.

pcan08 committed:
Merge branch 'dev' into area-in-state-content

pcan08 committed:
Create _hasArea method in HaEntityStatePicker

pcan08 committed:
Add device option in HaEntityStatePicker

pcan08 committed:
Add floor option in HaSelectorUiStateContent

pcan08 committed:
Refactor _has* methods

pcan08 committed:
Refactor in StateDisplay

pcan08 commented:
Hi !
I added floor and device.
Would you like that I add the indicator in the chips in case there is no area and floor for device when there are in the config but not set to the device/entity as proposed in the PR description ? 
Or do you prefer another PR that follows this one after merge (if merged) ?

pcan08 committed:
Use TemplateResult for new state content

pcan08 committed:
Optimize state display hierarchy resolution

  Only resolve entity→device→area→floor chain when displaying hierarchy-based content (area,
  device, floor), avoiding unnecessary registry lookups for other content types like
  timestamps and attributes.

pcan08 commented:
Thank you for the review, I'll deal with as soon as I can

pcan08 committed:
Remove name prefix

pcan08 committed:
Replace allow-* by allow-context (keep allow-name)

pcan08 committed:
Use name computation methods

pcan08 committed:
Fix remaning 'allow' attributes

pcan08 committed:
Use getEntityContext in picker

pcan08 committed:
Replace _resolveEntityHierarchy by getEntityContext in state-display

pcan08 committed:
Make bad config silent in card

Don't return TemplateResult avoid - char in card when there is unavailable/unknown value in state-content
